core/drand.go

```diff
@@ -92,19 +96,39 @@ func initDrand(s key.Store, c *Config) (*Drand, error) {
 	d.callbacks.AddCallback(callbackID, d.opts.callbacks)
 	//d.callbacks.AddCallback(cacheID, d.cache.StoreTemp)
 
-	a := c.ListenAddress(priv.Public.Address())
+	privAddr := c.PrivateListenAddress(priv.Public.Address())
+	pubAddr := c.PublicListenAddress("")
```

Comment:
this deserves a comment, since setting private address to `Public.Address()` looks scary